android/src/main/java/io/parity/signer/bottomsheets/KeyDetailsAction.kt

```diff
@@ -0,0 +1,83 @@
+package io.parity.signer.bottomsheets
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import io.parity.signer.alerts.AndroidCalledConfirm
+import io.parity.signer.components.BigButton
+import io.parity.signer.components.HeaderBar
+import io.parity.signer.models.*
+import io.parity.signer.ui.theme.Bg000
+import io.parity.signer.ui.theme.modal
+import io.parity.signer.uniffi.Action
+
+@Composable
+fun KeyDetailsAction(signerDataModel: SignerDataModel) {
+	var confirmForget by remember { mutableStateOf(false) }
+	var confirmExport by remember { mutableStateOf(false) }
+
+	Column(
+		Modifier.clickable { signerDataModel.navigate(Action.GO_BACK) }
+	) {
+		Spacer(Modifier.weight(1f))
+		Surface(
+			color = MaterialTheme.colors.Bg000,
+			shape = MaterialTheme.shapes.modal
+		) {
+			Column(
+				modifier = Modifier.padding(20.dp)
+			) {
+				HeaderBar(line1 = "KEY MENU", line2 = "Select action")
```

Comment:
I'm not aware of any advantages of using string resources except ability for localization. And there are downsides - it is more difficult to navigate through the code and finding proper files, responsible for UI element. 
Do you know any?

I would like to avoid using patterns purely because of patterns and extracting strings later will always be the same work. If we will ever consider localization.

Reply:
- keeping code DRY - if you reuse strings for some UI components, you can use just one unique id within static enum than copy same value all over
- imagine working with longer strings - would you inject 100 lines text within view or use just static reference provided by system to access it? 
- there is no downside to navigate, using sytem Resources mechanism provides you with mechanism to easily preview values in Android Studio AFAIK
- most of the language's linters would properly consider this as a code smell. I can imagine that we want to use some linter for Android codebase similar to iOS, in that case this would be a red flag
- it's not the same work - once you'll start copying those strings all over app is much harder to move them to single resource file then if you start with that file, right?
- consistency - you are utilising Resources for images, so using it for system designed intention like string makes sense?
- using keys instead of values for String is not just for localisation purposes - it's just a proper layer of concern separation - there is dedicated system component for UI-facing String management on given mobile platform (Resources Strings or Localizable strings on iOS) and it is its own responsibility to store this values in opposite to UI components that should just render **some value** based on given view model or keys to some resources

I understand downside when it comes to refactoring old UI components and dealing with that (although I must admit I went through that and it helps to organise old code too, but nvm), but for new components there is not more effort to use built-in resources comparing to just pasting string inline?

Reply:
1)I don't see an advantage of copying string id everewhere instead of string, it's still copying of the thing. Btw most strings are unique and used in 1 place only. At least if we won't mix titles with headers in another screen - those are different strings happened to have the same value.

2)At least in android we should never use 100 lines strings - it will take a lot to render. Independent if it's stored as stack value in code or in resources. And of cause big things I would extract - just to make code more readable.

This is old screen btw. Just git shoing it as created and removed from different folder.

3)Soddenly navigation doesn't work for preview. It's full text search if you need to fine where some text is used.

4)Well, android doesn't, at least by default. But you can put any rules you want to linter... It is a good practice to make app ready for localization so I understand why linter may have it on by default as well.

5)Often same value are not the same strings. All the cases where we decided change some generic value (in some languages there are 2 separate "cancel", for example, in meaning "back" and in meaning "abort") ended up to going through all the usages 1 by 1 anyway. I'm not taking about changing title in one screen that suddenly changing header in another - I think nobody using the same strings for those now days...

6)I don't see a consistency between images and strings. BTW we don't have images in android resources. It's only app icon. All current icons are taken from androidx library and defined in code
![image](https://user-images.githubusercontent.com/11879032/190368854-883910f6-ae5c-4fa3-9d5c-b074b0fccdbc.png)
But I wouldn't do it for our icons - I think for us readability is more important than runtime performance.

7)What is the difference between storing specific key and specific value right away? For case where we don't plan to support multiple languages.

8)And resources are less efficient at least from runtime performance point of view. Well, not in a meaning we will notice that, it's a good instrument for localization.  But when we don't plan localization I don't  see a need for instrument. The only reason I see is that somebody may fork the project and they do have plans for localisation - I can make it easier for them...

Reply:
Ok if it's important for you, I'll do that just so we'll be ready for translation that I hope we can do at some point with the help of FOSS community.